src/Sql/AbstractSql.php

```diff
@@ -241,14 +241,20 @@ protected function createSqlFromSpecificationAndParameters($specifications, $par
             if (isset($paramSpecs[$position]['combinedby'])) {
                 $multiParamValues = [];
                 foreach ($paramsForPosition as $multiParamsForPosition) {
-                    $ppCount = count($multiParamsForPosition);
-                    if (!isset($paramSpecs[$position][$ppCount])) {
-                        throw new Exception\RuntimeException(sprintf(
-                            'A number of parameters (%d) was found that is not supported by this specification',
-                            $ppCount
-                        ));
+                    if (is_array($multiParamsForPosition) or ($multiParamsForPosition instanceof Countable)) {
+                        $ppCount = count($multiParamsForPosition);
+                        if (!isset($paramSpecs[$position][$ppCount])) {
+                            throw new Exception\RuntimeException(sprintf(
+                                'A number of parameters (%d) was found that is not supported by this specification',
+                                $ppCount
+                            ));
+                        }
+                        $multiParamValues[] = vsprintf($paramSpecs[$position][$ppCount], $multiParamsForPosition);
+                    } else {
+                        throw new Exception\UnexpectedValueException(sprintf(
+                            'Parameters must be an array or a countable object, %s given',
+                            gettype($multiParamsForPosition)
```

Comment:
I think this should not raise an exception, as this may break some other code, for the  "NULL" case

At least in zend-paginator  test suite
```
1) ZendTest\Paginator\Adapter\DbTableGatewayTest::testGetItemsWithWhereAndOrderAndGroup
count(): Parameter must be an array or an object that implements Countable
/usr/share/php/Zend/Db/Sql/AbstractSql.php:244
/usr/share/php/Zend/Db/Sql/AbstractSql.php:72
/usr/share/php/Zend/Db/Sql/AbstractPreparableSql.php:34
/usr/share/php/Zend/Db/Sql/Platform/Platform.php:100
/usr/share/php/Zend/Db/Sql/Sql.php:130
/builddir/build/BUILDROOT/php-zendframework-zend-paginator-2.7.0-3.fc28.noarch/usr/share/php/Zend/Paginator/Adapter/DbSelect.php:100
/builddir/build/BUILD/zend-paginator-42211f3e1e8230953c641e91fec5aa9fe964eb95/test/Adapter/DbTableGatewayTest.php:121
2) ZendTest\Paginator\Adapter\DbTableGatewayTest::testGetItemsWithWhereAndOrderAndGroupAndHaving
count(): Parameter must be an array or an object that implements Countable
/usr/share/php/Zend/Db/Sql/AbstractSql.php:244
/usr/share/php/Zend/Db/Sql/AbstractSql.php:72
/usr/share/php/Zend/Db/Sql/AbstractPreparableSql.php:34
/usr/share/php/Zend/Db/Sql/Platform/Platform.php:100
/usr/share/php/Zend/Db/Sql/Sql.php:130
/builddir/build/BUILDROOT/php-zendframework-zend-paginator-2.7.0-3.fc28.noarch/usr/share/php/Zend/Paginator/Adapter/DbSelect.php:100
/builddir/build/BUILD/zend-paginator-42211f3e1e8230953c641e91fec5aa9fe964eb95/test/Adapter/DbTableGatewayTest.php:143

```